File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for ch in path.split('/'):
            if not ch or ch == '.':
                continue
            if ch != '..':
                stack.append(ch)
            elif stack:
                stack.pop()
        return '/' + '/'.join(stack)

File: 0071-simplify-path/0071-simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
  return '/' + path.split('/').reduce((acc, curr) => {
    if (!curr || curr === '.') return acc;
    
    curr === '..' ? acc.pop() : acc.push(curr);
    
    return acc;
  }, []).join('/');
};